arrow/src/record_batch.rs

```diff
@@ -412,16 +412,47 @@ pub struct RecordBatchOptions {
     /// Optional row count, useful for specifying a row count for a RecordBatch with no columns
     pub row_count: Option<usize>,
 }
+/// Builder to create new RecordBatchOptions object
+#[derive(Debug)]
+pub struct RecordBatchOptionsBuilder {
+    /// Match field names of structs and lists. If set to `true`, the names must match.
+    match_field_names: bool,
 
-impl Default for RecordBatchOptions {
+    /// Optional row count, useful for specifying a row count for a RecordBatch with no columns
+    row_count: Option<usize>,
+}
+impl Default for RecordBatchOptionsBuilder {
     fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl RecordBatchOptionsBuilder {
+    pub fn new() -> Self {
         Self {
             match_field_names: true,
             row_count: None,
         }
     }
+    pub fn match_field_names(mut self, match_field_names: bool) -> Self {
+        self.match_field_names = match_field_names;
+        self
+    }
+    pub fn row_count(mut self, row_count: usize) -> Self {
```

Comment:
```suggestion
    pub fn row_count(mut self, row_count: Option<usize>) -> Self {
```

I recommend having this parameter take an Option, otherwise there is no way to set it to `None`.

Reply:
The parameter is changed to `Option`